pydm/utilities/__init__.py

```diff
@@ -282,3 +288,17 @@ def _access_check(fn, mode):
                 if _access_check(name, mode):
                     return name
     return None
+
+
+def only_main_thread(func):
```

Comment:
Docstring should at least indicate that this is a decorator

Also verify `callable(func)` or raise